src/Extensions/ExtHostProtocol.re

```diff
@@ -280,8 +280,19 @@ module DiagnosticsCollection = {
     switch (json) {
     | `List([`String(name), `List(perFileDiagnostics)]) =>
       let perFileDiagnostics =
-        List.map(Diagnostics.of_yojson, perFileDiagnostics)
-        |> Utility.filterMap(Utility.resultToOption);
+        List.map(Diagnostics.of_yojson, perFileDiagnostics);
+
+      List.iter(
+        v =>
+          switch (v) {
+          | Ok(_) => ()
+          | Error(msg) => print_endline("ERROR: " ++ msg)
+          },
```

Comment:
This could be shortened a bit using `fun | ...`. That's what it's really for, I think. In 4.08 I think you can also use `Result.iter_error`.

Reply:
Oh, ya, good catch - I hadn't used the `fun` shorthand much but saw it in a few of your PRs. Indeed, it's much more streamlined:
```
v => switch(v) {
| ...
}
```
to: 
```
fun
| ....
```

I switched to use `fun`, but the `Result.iter_error` would be even more concise - might be worth bringing in a polyfilled `Result` module like we did for `Option`?